pytorch_lightning/trainer/trainer.py

```diff
@@ -989,6 +989,9 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
         # ----------------------------
         # TRAIN
         # ----------------------------
+        # reset logger connector
```

Comment:
It is easier to put these reset calls directly in the trainer but IMO they should be managed by each loop, even if it's more verbose.

Reply:
@awaelchli suggested this. It was in the loop before. Any reason why they should be inside loop?

Reply:
because the loops are the ones to control the flow of training, and they should be responsible for interacting with the logger connector, not the trainer.
Otherwise, the customization is limited.

For example, compare the existing usage of the `logger_connector` in the trainer vs the loops.

Reply:
well here is some advice 😅 
in most cases where @carmocca's arguments stand against @awaelchli's arguments, @carmocca wins. So when @awaelchli suggests changes better wait until @carmocca reviews before you make the changes 😄